packages/common-integration-tests/src/webhooks.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  PostUrlWebhookDestination,
+  ListWebhooks,
+  GetWebhookSecret,
+} from '@gomomento/sdk-core';
+import {
+  ItBehavesLikeItValidatesCacheName,
+  ItBehavesLikeItValidatesTopicName,
+  testWebhook,
+  ValidateCacheProps,
+  ValidateTopicProps,
+  WithWebhook,
+} from './common-int-test-utils';
+import {
+  ICacheClient,
+  ITopicClient,
+} from '@gomomento/sdk-core/dist/src/internal/clients';
+
+export function runWebhookTests(
+  topicClient: ITopicClient,
+  cacheClient: ICacheClient,
+  integrationTestCacheName: string
+) {
+  describe('#putWebhook', () => {
+    ItBehavesLikeItValidatesCacheName((props: ValidateCacheProps) => {
+      return topicClient.putWebhook({
+        id: {
+          cacheName: props.cacheName,
+          webhookName: 'ItBehavesLikeItValidatesCacheName',
+        },
+        topicName: 'a topic',
+        destination: new PostUrlWebhookDestination('someurl.com'),
+      });
+    });
+
+    ItBehavesLikeItValidatesTopicName((props: ValidateTopicProps) => {
+      return topicClient.putWebhook({
+        id: {
+          cacheName: 'a cache',
+          webhookName: 'ItBehavesLikeItValidatesCacheName',
+        },
```

Comment:
yeah, like, this looks a little odd that these two params aren't just at the same level as topicName and destination

Reply:
we can put them at the same level, but do you want to take them out of the object completely?

So sig becomes

`putWebhook(cacheName: string, topicName: string, webhookName: string, destination: string | WebhookDestination)`

Reply:
when there are that many strings i just worry that its too easy to swap like topicName and cacheName since typescript doesnt have named parameters

Reply:
yeah, i hear you - and if we were starting over from scratch you could probably sell me on making every API just have a single `options` or `props` argument to solve that. but since the existing APIs don't have that I feel like it would be inconsistent to do that just for this one / these, don't you?

Reply:
also I think all of the existing APIs in the current client start with `cacheName` and `topicName`, but after that, it's fair game. so e.g. look at the sig of `subscribe`. you could follow that pattern for webhooks and I think it would be consistent with the existing APIs.